dissect/target/loaders/uac.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from dissect.target import filesystem
+from dissect.target.filesystems.tar import TarFilesystemDirectoryEntry, TarFilesystemEntry
+from dissect.target.helpers import fsutil
+from dissect.target.loaders.dir import DirLoader, find_and_map_dirs, find_dirs, map_dirs
+from dissect.target.loaders.tar import TarSubLoader
+from dissect.target.loaders.zip import ZipSubLoader
+
+if TYPE_CHECKING:
+    import tarfile as tf
+    import zipfile as zf
+    from pathlib import Path
+
+    from dissect.target import Target
+
+log = logging.getLogger(__name__)
+
+FILESYSTEMS_ROOT = "[root]"
+UAC_CHECK_FILE = "uac.log"
+
+
+class UACLoader(DirLoader):
+    """UAC forensic image files (uncompressed or as .tar.gz or .zip).
+
+    References:
+        - https://github.com/tclahr/uac
+        - https://tclahr.github.io/uac-docs/
+    """
+
+    def __init__(self, path: Path, **kwargs):
+        super().__init__(path)
+
+    @staticmethod
+    def detect(path: Path) -> bool:
+        return path.joinpath(FILESYSTEMS_ROOT).exists() and path.joinpath(UAC_CHECK_FILE).exists()
+
+    def map(self, target: Target) -> None:
+        os_type, dirs = find_dirs(self.absolute_path.joinpath(FILESYSTEMS_ROOT))
+        map_dirs(target, dirs, os_type)
+
+
+class UacTarSubloader(TarSubLoader):
+    """Loader for tar-based UAC collections."""
+
+    FS_ROOT_TUPLE = (f"/{FILESYSTEMS_ROOT}/", f"{FILESYSTEMS_ROOT}/")
+
+    @classmethod
+    def detect(cls, tarfile: tf.TarFile) -> bool:
+        return any(member.name.startswith(cls.FS_ROOT_TUPLE) for member in tarfile.getmembers())
+
+    def map(self, target: Target) -> None:
+        vol = filesystem.VirtualFilesystem(case_sensitive=False)
+        vol.tar = self.tar
+        for member in self.tar.getmembers():
+            if member.name == ".":
+                continue
+
+            if member.name.startswith(self.FS_ROOT_TUPLE):
+                # Current acquire
+                parts = member.name.replace(f"{FILESYSTEMS_ROOT}/", "").split("/")
+                if parts[0] == "":
+                    parts.pop(0)
+
+                mname = "/".join(parts)
+                entry_cls = TarFilesystemDirectoryEntry if member.isdir() else TarFilesystemEntry
+                entry = entry_cls(vol, fsutil.normpath(mname), member)
+                vol.map_file_entry(entry.path, entry)
+        target.filesystems.add(vol)
+
+
+class UacZipSubLoader(ZipSubLoader):
+    """Loader for zip-based UAC collections."""
+
+    @staticmethod
+    def detect(zipfile: zf.Path) -> bool:
+        return zipfile.joinpath(FILESYSTEMS_ROOT).exists() and zipfile.joinpath(UAC_CHECK_FILE).exists()
+
+    def map(self, target: Target) -> None:
+        path = self.zip
+        if path.joinpath(FILESYSTEMS_ROOT).exists():
+            path = path.joinpath(FILESYSTEMS_ROOT)
```

Comment:
```suggestion
        path = self.zip.joinpath(FILESYSTEMS_ROOT)
```
We should only get here if this is already true (due to the detect logic) so I think this can be simplified.